File: week-02/day-02/diagonal_matrix.py
# - Create (dynamically) a two dimensional list
#   with the following matrix. Use a loop!
#
#   1 0 0 0
#   0 1 0 0
#   0 0 1 0
#   0 0 0 1
#
# - Print this two dimensional list to the output

# a = [[0 for y in range(4)] for x in range(4)]

#a = []
#for x in range(8):
#    row = []
#    for y in range(8):
#        row.append(0)
#    a.append(row)

a = []
for x in range(4):
    row = []
    for y in range(4):
        if y == x:
            row.append(1)
        else:
            row.append(0)
    a.append(row)
print(a)
print("\n")

oneline = ""
for j in range(4):
    for i in range(4):
        oneline += str(a[i][j])
        oneline += " "
    print(oneline)
    oneline = ""

File: week-07/day-02/diagonal-matrix.js
'use strict';
// - Create (dynamically*) a two dimensional list
//   with the following matrix**. Use a loop!
//
//   0 0 0 1
//   0 0 1 0
//   0 1 0 0
//   1 0 0 0
//
// - Print this two dimensional list to the console
//
// * size should depend on a variable
// ** Relax, a matrix is just like an array

let size = 4;
let matrix = [];

for (let i = 0; i < size; i++) {
  matrix[i] = [];
  for (let j = 0; j < size; j++) {
    if (i + j === size - 1) {
      matrix[i][j] = 1;
    } else {
      matrix[i][j] = 0;
    }
  }
}
let line = '';
for (let i = 0; i < size; i++) {
  for (let j = 0; j < size; j++) {
    line += matrix[i][j] + ' ';
  }
  console.log(line);
  line = '';
}
